lib/babe/verify.go

```diff
@@ -50,7 +52,14 @@ func NewVerificationManager(blockState BlockState, epochState EpochState) (*Veri
 		return nil, ErrNilBlockState
 	}
 
-	if epochState == nil {
+	var isNilEpoch bool
+	switch epochState.(type) {
+	case *state.EpochState:
+		isNilEpoch = epochState == (*state.EpochState)(nil)
+	default:
+		isNilEpoch = reflect.ValueOf(epochState).IsNil()
+	}
+	if isNilEpoch {
```

Comment:
not a performance critical path, so let's just use for all:

```suggestion
	if reflect.ValueOf(epochState).IsNil() {
```

Reply:
Updated.